Enterprise-CMCS/macpro-mdct-mcr
# AWS SDK V3 migration

### Description


Upgrade MCR to use AWS SDK v3

This PR was greatly assisted by the efforts of @benmartin-coforma in the [upgrade for CARTS](https://github.com/Enterprise-CMCS/macpro-mdct-carts/pull/139521)

Tips for reviewing:
- Start with everything other than app-api
- Once you get to app-api start with package.json, then the utils/ folder
- Almost every change to a handler is updating how it uses the utils and new types
- Almost every change to a test is updating mocks for the new sdk

Notes:
- s3 get is different https://docs.aws.amazon.com/AmazonS3/latest/userguide/example_s3_GetObject_section.html
  - it now returns a streamed body, which we have to convert to a string and then parse. this is updated in the s3-lib with virtually no impact elsewhere (except mocks in tests)

References:
- AWS SDK v3 [docs](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/)

### Related ticket(s)


CMDCT-3074

---
### How to test

deployed url: https://d1jtat55rts1jz.cloudfront.net/

Click around and make sure everything is ok!

Our cypress tests cover everything that was changed here, so you can just validate whatever use cases you like.

### Important updates


yarn everything

---
### Author checklist


---

### Feedback

I'm interested in feedback on better ways to handle the test mocking. They resulted in a lot of duplication, though I don't want to mock too globally for fear of having false positives.
